### PR TITLE
gptjudge empty response handling

### PR DESCRIPTION
**Pull request by snova-nidhih**
Occasionally when using gpt as a judge, the response is empty. From our test set, this occurred only once and is considered incorrect by default. 

Test command:
`python3 -m accelerate.commands.launch \
    --num_processes=1 \
    -m lmms_eval \
    --model internvl2 \
    --model_args pretrained="/import/ml-sc-scratch3/nidhih/mm_hungarian/finetune_outputs/pretrain_hupdf2_synthdog_hu_all_unfrozen" \
    --tasks docvqa_hu_syn \
    --batch_size 1 \
    --log_samples \
    --log_samples_suffix intervl2_8b \
    --output_path /import/ml-sc-scratch3/nidhih/mm_hungarian/lmms_eval_op/internvl2_8b_docvqasyn_pt_test \
    --limit 200`

### CODE REVIEW

File: lmms_eval/api/metrics.py [hunk 1]
```diff
@@ -405,7 +405,10 @@ def gpt4judge(references, predictions, query):  # This is a passthrough function
                     eval_logger.error(f"All 5 attempts failed. Last error message: {str(e)}.\nResponse: {str(error_msg)}")
                     response = ""
 
-        score = int(extract_number_from_brackets(response))
+        if response is None:  # Rare case of gpt returning empty response
+            score = 0
```
Comment (snova-jonathanl) [inline lmms_eval/api/metrics.py:409]: Can we do a try+except wrapping `extract_number_from_brackets`, and also add logging whenever this is caught?
